embedchain/pipeline.py

```diff
@@ -346,22 +342,40 @@ def deploy(self):
         self.telemetry.capture(event_name="deploy", properties=self._telemetry_props)
 
     @classmethod
-    def from_config(cls, yaml_path: str, auto_deploy: bool = False):
+    def from_config(cls, yaml_path: Optional[str] = None, auto_deploy: bool = False, **kwargs: Dict[str, Any]):
         """
         Instantiate a Pipeline object from a YAML configuration file.
 
         :param yaml_path: Path to the YAML configuration file.
         :type yaml_path: str
         :param auto_deploy: Whether to deploy the pipeline automatically, defaults to False
         :type auto_deploy: bool, optional
+        :param kwargs: Additional keyword arguments.
+        :type kwargs: Dict[str, Any]
+            - json_path: Path to the JSON configuration file.
+            - config: A dictionary containing the configuration.
         :return: An instance of the Pipeline class.
         :rtype: Pipeline
         """
-        with open(yaml_path, "r") as file:
-            config_data = yaml.safe_load(file)
+        json_path = kwargs.get("json_path", None)
+        config = kwargs.get("config", None)
+
+        if yaml_path:
```

Comment:
I think you have to check before this if condition to make sure that user passes either of three `yaml_path`, `json_path`, `config` in this function. Error out if user passes any two or three.